processing/src/main/java/io/druid/query/search/SearchQueryQueryToolChest.java

```diff
@@ -144,15 +161,13 @@ public SearchQueryQueryToolChest(
         final byte[] querySpecBytes = query.getQuery().getCacheKey();
         final byte[] granularityBytes = query.getGranularity().cacheKey();
 
-        final Collection<DimensionSpec> dimensions = query.getDimensions() == null
-                                                     ? ImmutableList.<DimensionSpec>of()
-                                                     : query.getDimensions();
-
-        final byte[][] dimensionsBytes = new byte[dimensions.size()][];
+        final List<DimensionSpec> dimensionSpecs =
```

Comment:
the `query` parameter is kind of silly anyway (it's always the same as the one used to create the strategy) so I think it's ok to use it, or ignore it, whatever is more convenient.